src/plugins/intel_cpu/src/transformation_pipeline.cpp

```diff
@@ -571,10 +570,34 @@ void Transformations::MainSnippets(void) {
     if (!isMHASupported) {
         snippetsManager.get_pass_config()->disable<ngraph::snippets::pass::TokenizeMHASnippets>();
     }
+
+    auto is_supported_matmul = [](const std::shared_ptr<const ov::op::v0::MatMul>& matmul) {
+        if (!matmul)
+            return false;
+        if (matmul->get_input_element_type(1) == ov::element::i8)
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_vnni);
+        if (matmul->get_input_element_type(0) == ov::element::bf16 &&
+            matmul->get_input_element_type(1) == ov::element::bf16)
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16);
+        return true;
+    };
+
     if (snippetsMode != Config::SnippetsMode::IgnoreCallback) {
         snippetsManager.get_pass_config()->set_callback<ngraph::snippets::pass::TokenizeMHASnippets>(
-                [](const std::shared_ptr<const ov::Node>& n) -> bool {
-                    const auto pshape = n->get_output_partial_shape(0);
+                [this, is_supported_matmul](const std::shared_ptr<const ov::Node>& n) -> bool {
+                    if (this->enableLpt) {
+                        // Tranformation callback is called on MatMul1
+                        if (!is_supported_matmul(ov::as_type_ptr<const ov::op::v0::MatMul>(n)))
```

Comment:
Minor: it makes a little sense to cast here. imagine the cast is not successful: we pass nullptr to the lambda and check it there simply to return false, but we already knew that we're passing nullptr (it 100% can't be supported matmul)
suggestion: why don't we pass `Node` ptr directly to the lambda, so we can both cast and check there?